decode/clarity.ts

```diff
@@ -46,83 +56,109 @@ export function decode(input: string | IPayload, augmentations: IAugmentation =
             case Event.TouchCancel:
             case Event.TouchEnd:
             case Event.TouchMove:
-                event = interaction(entry);
-                payload.analytics.push(event);
+                if (payload.pointer === undefined) { payload.pointer = []; }
+                payload.pointer.push(interaction.decode(entry) as Decode.PointerEvent);
+                break;
+            case Event.Scroll:
+                if (payload.scroll === undefined) { payload.scroll = []; }
+                payload.scroll.push(interaction.decode(entry) as Decode.ScrollEvent);
+                break;
+            case Event.Resize:
+                if (payload.resize === undefined) { payload.resize = []; }
+                payload.resize.push(interaction.decode(entry) as Decode.ResizeEvent);
+                break;
+            case Event.Selection:
+                if (payload.selection === undefined) { payload.selection = []; }
+                payload.selection.push(interaction.decode(entry) as Decode.SelectionEvent);
+                break;
+            case Event.Change:
+                if (payload.change === undefined) { payload.change = []; }
+                payload.change.push(interaction.decode(entry) as Decode.ChangeEvent);
                 break;
             case Event.BoxModel:
-                event = layout.decode(entry);
-                payload.playback.push(event);
+                if (payload.boxmodel === undefined) { payload.boxmodel = []; }
+                payload.boxmodel.push(layout.decode(entry) as Decode.BoxModelEvent);
                 break;
             case Event.Discover:
             case Event.Mutation:
-                event = layout.decode(entry);
-                payload.playback.push(event);
+                if (payload.dom === undefined) { payload.dom = []; }
+                payload.dom.push(layout.decode(entry) as Decode.DomEvent);
                 break;
             case Event.Checksum:
```

Comment:
>Checksum [](start = 23, length = 8)

CAn we rename this to LayoutSummary or something more indicative? "Hash" would be better than "Checksum" (which usually is a check for errors in the data).

Reply:
Renamed it to Hash